src/coap_options.h

```diff
@@ -41,7 +42,7 @@ typedef enum
 
 //on addition of an option also change coap_options.C (array initializer) and count
 //Proxy-Uri
-#define KNOWN_OPTIONS_COUNT (6)
+#define KNOWN_OPTIONS_COUNT (7)
 extern uint16_t KNOWN_OPTIONS[KNOWN_OPTIONS_COUNT];
```

Comment:
`KNOWN_OPTIONS_COUNT` could be defined based on `sizeof()` to avoid counting the elements manually.

```
#define LEN(x)  (sizeof(x) / sizeof((x)[0]))
```

or just `#define KNOWN_OPTIONS_COUNT (sizeof(KNOWN_OPTIONS) / sizeof(KNOWN_OPTIONS[0]))`

or similar, whatever you like more :)